File: backend/src/repository/dto/create-repository.dto.ts
import { Matches, IsNotEmpty, IsOptional, IsString } from 'class-validator';

export class CreateRepositoryDto {
  @IsNotEmpty()
  @Matches(/^(https?:\/\/|git@|ssh:\/\/)([^\s]+)$/, { 
    message: 'Please provide a valid Git URL (HTTPS or SSH)' 
  })
  url: string;

  @IsOptional()
  @IsString()
  name?: string;

  @IsNotEmpty()
  @IsString()
  branch: string;

  @IsOptional()
  @IsString()
  subPath?: string;

  @IsOptional()
  @IsString()
  sshKeyId?: string;

  @IsOptional()
  scanIntervalMinutes?: number;

  @IsOptional()
  @IsString()
  scanCron?: string;
}

File: backend-java/src/main/java/com/asmolabs/zanshin/repository/dto/CreateRepositoryDto.java
package com.asmolabs.zanshin.repository.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

public record CreateRepositoryDto(
    @NotBlank(message = "Please provide a valid Git URL")
    @Pattern(regexp = "^(https?://|git@|ssh://)([^\\s]+)$", message = "Please provide a valid Git URL (HTTPS or SSH)")
    String url,

    String name,

    @NotBlank(message = "Branch is required")
    String branch,

    String subPath,

    String sshKeyId,

    Integer scanIntervalMinutes,

    String scanCron
) {}
